akka-actor/src/main/scala/akka/actor/AbstractActor.scala

```diff
@@ -5,17 +5,74 @@
 package akka.actor
 
 import akka.annotation.ApiMayChange
+import akka.japi.pf.ReceiveBuilder
+import akka.japi.pf.UnitPFBuilder
+import scala.runtime.BoxedUnit
 
 /**
  * Java API: compatible with lambda expressions
- *
- * This is an EXPERIMENTAL feature and is subject to change until it has received more real world testing.
  */
 object AbstractActor {
+
+  /**
+   * Defines which messages the Actor can handle, along with the implementation of
+   * how the messages should be processed. You can build such behavior with the
+   * [[akka.japi.pf.receivebuilder]] but it can be implemented in other ways than
+   * using the `ReceiveBuilder` since it in the end is just a wrapper around a
+   * Scala `PartialFunction`.
+   */
+  final class Receive(val onMessage: PartialFunction[Any, BoxedUnit]) {
+    /**
+     * Composes this `Receive` with a fallback which gets applied
+     * where this partial function is not defined.
+     */
+    def orElse(other: Receive): Receive = new Receive(onMessage.orElse(other.onMessage))
+  }
+
   /**
    * emptyBehavior is a Receive-expression that matches no messages at all, ever.
    */
-  final val emptyBehavior = Actor.emptyBehavior
+  final val emptyBehavior: Receive = new Receive(PartialFunction.empty)
+
+  /**
+   * The actor context - the view of the actor cell from the actor.
+   * Exposes contextual information for the actor and the current message.
+   */
+  trait ActorContext extends akka.actor.ActorContext {
+
+    /**
+     * Returns an unmodifiable Java Collection containing the linked actors,
+     * please note that the backing map is thread-safe but not immutable
+     */
+    def getChildren(): java.lang.Iterable[ActorRef]
+
+    /**
+     * Returns a reference to the named child or null if no child with
+     * that name exists.
+     */
+    def getChild(name: String): ActorRef
```

Comment:
It's a scary one indeed. OTOH, it would be a really nice cleanup. Personally, I think I would bite the bullet and change it (if we can w.r.t. non-experimental API promises).

Maybe do a little poll on the chat?